be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -553,16 +559,35 @@ Status SegmentIterator::_init_column_iterator_by_cid(const ColumnId cid, const C
                 file_size = rfile->get_size().value();
             }
 
-            auto shared_buffered_input_stream =
-                    std::make_unique<io::SharedBufferedInputStream>(rfile->stream(), _segment->file_name(), file_size);
-            const io::SharedBufferedInputStream::CoalesceOptions options = {
-                    .max_dist_size = config::io_coalesce_read_max_distance_size,
-                    .max_buffer_size = config::io_coalesce_read_max_buffer_size};
-            shared_buffered_input_stream->set_coalesce_options(options);
-            iter_opts.read_file = shared_buffered_input_stream.get();
-            iter_opts.is_io_coalesce = true;
-            _column_files[cid] = std::move(shared_buffered_input_stream);
-            _io_coalesce_column_index.emplace_back(cid);
+            // if the size of segment file is small, read the whole file directly
+            if (file_size <= config::lake_small_segment_file_threshold_size) {
+                if (_shared_buffer_input_stream == nullptr) {
+                    _shared_buffer_input_stream = std::make_unique<io::SharedBufferedInputStream>(
+                            rfile->stream(), _segment->file_name(), file_size);
+                    const io::SharedBufferedInputStream::CoalesceOptions options = {
+                            .max_dist_size = config::io_coalesce_read_max_distance_size,
+                            .max_buffer_size = config::io_coalesce_read_max_buffer_size};
+                    _shared_buffer_input_stream->set_coalesce_options(options);
+
+                    // read the entire file at once by set range{0, file_size}
+                    std::vector<io::SharedBufferedInputStream::IORange> ranges = {{0, file_size}};
+                    RETURN_IF_ERROR(_shared_buffer_input_stream->set_io_ranges(ranges));
+                }
+                iter_opts.read_file = _shared_buffer_input_stream.get();
+
```

Comment:
Wouldn't this logic also work in shared nothing mode? If so, does it work correctly? Also, if so, change the name of the configuration entry `lake_small_segment-file_threshold_size`